test/unit/math/mix/prob/std_normal_cdf_log_test.cpp

```diff
@@ -0,0 +1,20 @@
+#include <test/unit/math/test_ad.hpp>
+#include <limits>
+
+TEST(mathMixScalFun, std_normal_lcdf_derivatives) {
+  auto f = [](const auto& y) { return stan::math::std_normal_lcdf(y); };
+
+  stan::test::expect_ad(f, -50.0);
```

Comment:
Can you do these tests for std_normal_lpdf, std_normal_lcdf and std_normal_lccdf too?

Then we can get rid of some other tests and make sure we're staying up to date with the testing framework stuff.

Reply:
This test is for `std_normal_lcdf` already, but uses the existing (albeit confusing) naming convention of calling it `std_normal_cdf_log_test.cpp`. I'm adding mix tests for `_cdf`, `_lccdf` and `_lpdf`.

Reply:
In the `_cdf` test I'm seeing these failures that I'm not sure if I should worry about: for `stan::test::expect_ad(f, -25.0)` I get:
```
./test/unit/math/expect_near_rel.hpp:92: Failure
Value of: stan::math::is_nan(x1) && stan::math::is_nan(x2)
  Actual: false
Expected: true
expect_near_rel(4.7760521554589183e-134, -nan)
expect_near_rel; require items x1(i) = x2(i): grad_hessian() grad Hessian
```
while for `stan::test::expect_ad(f, -20.0)`:
```
./test/unit/math/expect_near_rel.hpp:96: Failure
Expected equality of these values:
  x1
    Which is: 2.20286e-85
  x2
    Which is: inf
expect_near_rel(2.2028583965234379e-85, inf)
```

Reply:
The first doesn't really bother me that much cuz it's saying our numerics aren't perfect 25 standard deviations away from the mean on the non-log scale.

The second though... That's weird. The number on the left seems more right.

@bob-carpenter for at least one Q -- Is x1 or x2 the reference?

Reply:
I think x1 is finite differences, x2 is ad. The message is about `grad_hessian() grad Hessian` (sorry I cut off the last line). The test passes at -19.1 and starts failing at -19.2 with message `expect_near_rel(1.3097063068172137e-78, inf)`. Note that also `normal_cdf` is affected similarly, only it doesn't have such test.

Reply:
Oooh, okay. Yeah if it's with 2nd derivatives that doesn't bother me. I thought it was values. That's fine.